lib/src/ovis_log/ovis_log.c

```diff
@@ -945,6 +940,8 @@ int ovis_vlog(ovis_log_t log, int level, const char *fmt, va_list ap)
 		/* No workers, so directly log to the file. */
 		rc = __log(log, level, msg, &tv, &tm);
 		free(msg);
+		if (log_fp != OVIS_LOG_SYSLOG)
```

Comment:
I don't understand this condition.

Reply:
The if-statement prevents calling `fflush()` on an invalid `FILE *` stream, which is `OVIS_LOG_SYSLOG` (0x7). 

When `ovis_log_open()` is called with `syslog` as the path, I set `log_fp` to `OVIS_LOG_SYSLOG`. After investigation, I believe a better approach would be to let applications specify syslog output during initialization through the `modes` argument of the `ovis_log_init()` function. Since `modes` is a mask, we could add a new flag for syslog support. Then, the if-statement will check if the syslog mode is enabled.

The patch is going into b4.4. If we agree on chaning the syslog handling in ovis_log, I could implement the change for the top of tree.